.github/workflows/ci-test-library.yml

```diff
@@ -35,20 +35,29 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Rust latest
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+
+      - name: rust latest
         run: rustup update
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
+      # check how to use virtual environment https://stackoverflow.com/q/74668349/4001592
+      - name: create virtual environment
+        run: python -m venv .venv
 
-      - name: Poetry install
-        run: poetry install
-        working-directory: ${{ github.workspace }}
+      - name: install dependencies
+        run: .venv/bin/python -m pip install pytest maturin
```

Comment:
@mesejo be aware that directly running a binary in the venv without activation will have issues if python needs access to any of the binaries provided in `.venv/bin`